Source/WebKit/UIProcess/API/haiku/WebViewBase.cpp

```diff
@@ -84,8 +84,17 @@ void WebViewBase::Draw(BRect update)
         return;
 
     IntRect updateArea(update);
-    WebCore::Region unpainted;
-    drawingArea->paint(this, updateArea, unpainted);
+
+    callOnMainThread([this, drawingArea, updateArea](){
+        LockLooper();
+        WebCore::Region unpainted;
+
+        // TODO: make paint support multithreading. That way we don't have to
+        // call this on the main thread.
```

Comment:
things happening in the main thread is a quite central thing in WebKit and probably hard to change.